<a href="https://colab.research.google.com/github/Suzzi-e/data_visualisation/blob/main/Karamoja_Project.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

## Project Submssion
* Student name: Susan Akinyi Ochieng
* Student pace: Part-Time
* Date: 01/09/2024
* Instructor name: Samwel Jane/ Veronica isiaho



# Karamoja mockup of the Food Security Monitoring Tool

What or where is Karamoja? Karamoja is the most food-inecure region in Uganda. Therefore, several NGOs provide technical support as well as farm inputs to the farmers experiencing extremely low yield.Though, they lack visibility into the overall state of the region, they often need to rely on some very local sources of information to prioritise their activities.

## Research Question
The agri-tech team is asking you to develop an interactive visualisation tool of the results of the first crop season. This visualisation tool that you will develop will be used as a first mockup of the Food Security Monitoring tool that
Dalberg Data Insights (DDI) which is our local source of information will develop for the NGO.

## Objectives
* Develop a visualisation tool that will be used as a first mockup of the Food Security Monitoring tool that DDI will develop for the NGO.
* Visualise the results by district or sub-county (two administrative levels used by the NGO)

Now before we even get to the visualisations, which will be done on Tableau, we need to perform data cleaning on the dataset we will be working with and familiarise ourselves with the content. In this notebook we will perform data cleaning then proceed to Tableau to connect the datasets for visualisation. The datasets have been obtained from https://moringa.instructure.com/courses/727/assignments/52410?module_item_id=112659 .

## Data Cleaning

In [1]:
# Importing to pandas to load the csv in pandas
import pandas as pd
# Loading the file
df_subcounty = pd.read_csv('/content/Uganda_Karamoja_Subcounty_Crop_Yield_Population.csv')
# Using .Info() to get a concise summary of the dataframe's structure
df_subcounty.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 52 entries, 0 to 51
Data columns (total 13 columns):
 #   Column          Non-Null Count  Dtype  
---  ------          --------------  -----  
 0   OBJECTID        52 non-null     int64  
 1   SUBCOUNTY_NAME  52 non-null     object 
 2   DISTRICT_NAME   52 non-null     object 
 3   POP             52 non-null     int64  
 4   Area            52 non-null     int64  
 5   Karamoja        52 non-null     object 
 6   S_Yield_Ha      52 non-null     float64
 7   M_Yield_Ha      52 non-null     float64
 8   Crop_Area_Ha    52 non-null     float64
 9   S_Area_Ha       52 non-null     float64
 10  M_Area_Ha       52 non-null     float64
 11  S_Prod_Tot      52 non-null     float64
 12  M_Prod_Tot      52 non-null     float64
dtypes: float64(7), int64(3), object(3)
memory usage: 5.4+ KB


From the output above, The dataset has no missing values hence the handling of missing values step in data cleaning wil be skipped. We can view the data to know our next step.

In [2]:
# Viewing the first rows of our dataset using .head()
df_subcounty.head()

OBJECTID       SUBCOUNTY_NAME DISTRICT_NAME    POP        Area Karamoja  \
0       263              KACHERI        KOTIDO  17244  1067176155        Y   
1       264               KOTIDO        KOTIDO  52771   597575188        Y   
2       265  KOTIDO TOWN COUNCIL        KOTIDO  27389    23972401        Y   
3       266         NAKAPERIMORU        KOTIDO  38775   419111591        Y   
4       267           PANYANGARA        KOTIDO  65704   880955930        Y   

   S_Yield_Ha   M_Yield_Ha  Crop_Area_Ha     S_Area_Ha   M_Area_Ha  \
0  354.207411  1137.467019   7023.533691   6434.342449  528.124229   
1  367.890523  1162.996687  13587.990760  12455.592640  824.767081   
2  369.314177  1167.005832   1656.531855   1520.322052    8.561644   
3  283.324569   852.366578   7087.823334   6761.488901   45.721712   
4  373.836926  1283.859882  10398.249390  10111.198130  172.611914   

     S_Prod_Tot     M_Prod_Tot  
0  2.279092e+06  600723.892900  
1  4.582294e+06  959201.382500  
2  5.614765e+05    9991.488268  
3  1.915696e+06   38971.659080  
4  3.779939e+06  221609.511400

In [3]:
# Viewing the last rows of our dataset using .tail()
df_subcounty.tail()

OBJECTID SUBCOUNTY_NAME  DISTRICT_NAME    POP        Area Karamoja  \
47      1029      NGOLERIET          NAPAK  20790   158887552        Y   
48      1295        NYAKWAE           ABIM  16470   769609687        Y   
49      1313         LOKOPO          NAPAK  23200  1794470536        Y   
50      1318           RUPA         MOROTO  41493  2069554899        Y   
51      1320        MORUITA  NAKAPIRIPIRIT  16588   839293722        Y   

    S_Yield_Ha   M_Yield_Ha  Crop_Area_Ha    S_Area_Ha   M_Area_Ha  \
47  134.084089   528.699822   1215.227445  1145.339875   68.124005   
48  329.759030   779.225031    792.898789   342.816838  331.111340   
49  120.862232   748.829862   6471.047334  5830.549392  553.543123   
50  114.270921   699.334309   2217.290717  1989.119508  222.838881   
51  430.557375  1248.955812   1161.390229   185.283445  959.671162   

      S_Prod_Tot    M_Prod_Tot  
47  153571.85410  3.601715e+04  
48  113046.94800  2.580102e+05  
49  704693.21440  4.145096e+05  
50  227298.51840  1.558389e+05  
51   79775.15368  1.198587e+06

From the output of .info() the data types of the last 7 columns are in float but since we opened the csv by reading it, The data types of these columns cannot be altered unless we create a copy. Correcting the data types (Another step in data cleaning) will not be necessary here because the data type float still allows us to obtain stastistical values of these columns.

We can check for duplicates another step in data cleaning. This has been done in the cell below.




In [4]:
# Checking for duplicates
df_subcounty.duplicated().value_counts()

False    52
Name: count, dtype: int64

The output shows that are there no duplicates. We can then proceed to perform statistical measures on our dataset. The Pandas library has a built in function ```.describe(```) that is suitable for this step.

## Statistical Measures

In [5]:
# Using .describe() to obtain statistical values
df_subcounty.describe()

OBJECTID            POP          Area  S_Yield_Ha   M_Yield_Ha  \
count    52.000000      52.000000  5.200000e+01   52.000000    52.000000   
mean    787.865385   28934.692308  5.331913e+08  274.165405   940.259552   
std     280.101314   20865.122974  4.913308e+08  118.569907   321.641901   
min     263.000000    1418.000000  2.121209e+06  108.156411     0.000000   
25%     597.750000   16558.500000  1.568923e+08  173.034066   743.075879   
50%     810.500000   23053.500000  3.848356e+08  277.255206  1016.684002   
75%     982.250000   39461.000000  7.749029e+08  368.246437  1203.548665   
max    1320.000000  100919.000000  2.069555e+09  560.313070  1396.991494   

       Crop_Area_Ha     S_Area_Ha    M_Area_Ha    S_Prod_Tot    M_Prod_Tot  
count     52.000000     52.000000    52.000000  5.200000e+01  5.200000e+01  
mean    2839.646974   2253.143395   536.300569  6.557443e+05  5.500730e+05  
std     3110.505917   2954.355858   724.092288  9.915839e+05  7.939707e+05  
min        0.171390      0.130941     0.000000  1.728126e+01  0.000000e+00  
25%      964.876031    405.394759    79.821743  1.210555e+05  6.087012e+04  
50%     1654.265138   1231.824456   326.479336  2.543687e+05  2.896239e+05  
75%     3267.564651   2429.985069   740.296675  6.040942e+05  8.114574e+05  
max    13587.990760  12964.499730  3840.698081  4.582294e+06  4.365058e+06

## Findings

From the output above looking at mean (Rounded off the values to 2 decimal places):

* Yields: Maize did better than sorghum with a mean of 940.26 and 274.17 respectively.
* Area: Sorghum covers a greater area than Maize with a mean of 2253.14 and 536.30 respectively.
* Production: Sorghum has a higher production than that of Maize.

## Conclusions

* Maize yields better than sorghum by a huge gap.
* The area covered by Sorghum is vast compared to that of Maize.
* Sorghum has a higher produce than Maize.

## Recommendations

* Increased area coverage of maize since it yields better than Sorghum also the area covered by maize is little.
* Increased farm inputs on Sorghum fields to increase yields.